### DISCUSSION TITLE
Enabling revisions requires minimum two revisions to get subscriptions working with both previous and current

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#15900] [Q&A] [answered]
We need revisions enabled in our subscriptions so we can get notified when deletes happen. One strange thing is that if we only have one revision in our config, previous is always `null`. Is this is bug?

**Comment by ayende** [comment, marked as answer]
Not a bug, it is by design. When you create a new document, we have the first revision.
You'll get that as `Current = value, Previous = null`, indicating that this is a new document.
In the same sense, `Current = null, Previous = value` means that this is a delete.

Note that a delete and re-creation of the document would show as separate events here.
